src/vss_tools/vspec/vssexporters/vss2samm/helpers/ttl_helper.py

```diff
@@ -0,0 +1,190 @@
+    )
+
+    # Skip parsing of VSSNode is branches, which have only deprecated children
+    # NOTE: this is a special case for the OBD branch, which children are marked as deprecated from VSS 5.0,
+    #       but the branch itself is not marked as deprecated.
+    deprecated_children = []
+    if hasattr(vss_node.data, "type") and vss_node.data.type is NodeType.BRANCH:
+        for child_node in vss_node.children:
+            if (
+                hasattr(child_node.data, "deprecation")
+                and child_node.data.deprecation
+                and len(child_node.data.deprecation.strip()) > 0
+            ):
+                deprecated_children.append(child_node)
+
+    if len(deprecated_children) == len(vss_node.children):
+        log.warning(
+            "All child nodes of VSSNode: '%s' are deprecated.\n" "Skip the parsing of VSSNode: '%s'.\n",
+            vss_node.name,
+            vss_node.name,
+        )
+
+        return "DEPRECATED"
+
+    # Initialize RDF Graph for current node
+    graph = ttl_builder.setup_graph()
+
+    # Build VSS graph tree node.
+    # NOTE: ESMF-AME requires standalone aspect models to have an aspect node.
+    node_uri = handle_vss_node(path_to_ttl, graph, vss_node, True, split_vss)
+
+    if node_uri != "DEPRECATED":
+        # Print graph for current vss_node to a TTL file
+        # NOTE: make sure that TTL file name will reflect this graph's Aspect model name, i.e. uc first camel case
+        vss_node_ttl_file = write_graph_to_file(path_to_ttl, str_to_uc_first_camel_case(vss_node.ttl_name), graph)
+
+        log.debug("TTL file for parsed VSS node: '%s' is:\n'%s'\n", vss_node.name, vss_node_ttl_file)
+
+    return node_uri
+
+
+def handle_vss_node(path_to_ttl: Path, graph: Graph, vss_node: VSSNode, is_aspect: bool, split_vss: bool):
+    log.debug(
+        "Handle VSSNode: '%s'\n  -- node VSS path: '%s'\n  -- is_aspect: '%s'\n  -- is_expanded: '%s'\n",
+        vss_node.name,
+        vss_node.get_fqn(),
+        is_aspect,
+        vss_helper.is_node_expanded(vss_node),
+    )
+
+    # Check if node is deprecated and return DEPRECATED so it will be skipped for further conversion to TTL
+    if (
+        hasattr(vss_node.data, "deprecation")
+        and vss_node.data.deprecation
+        and len(vss_node.data.deprecation.strip()) > 0
+    ):
+        log.warning(
+            "Skipping VSSNode: '%s' since it is deprecated.\nDeprecation: '%s'\n",
+            vss_node.name,
+            vss_node.data.deprecation,
+        )
+
+        return "DEPRECATED"
+
+    # Build the general graph node for current vss_node
+    node_uri = ttl_builder.add_graph_node(graph, vss_node, is_aspect)
+
+    if hasattr(vss_node.data, "type") and vss_node.data.type is NodeType.BRANCH:
+        node_char_uri = ttl_builder.add_node_branch_characteristic(graph, vss_node, node_uri)
+
+        if (
+            hasattr(vss_node.data, "instances")
```

Comment:
if it is a branch it will always have instances filled therefore `if vss_node.data.instances` should  suffice

Reply:
I was getting similar to previous screenshot VSSRaw error if I've removed the hasattr check

Reply:
Done.

I updated the checks as advised.